docs/source/cli/model-training-inference/real-time-inference.rst

```diff
@@ -0,0 +1,320 @@
+critical for real-time infernce, it is recommened to use OLTP graph database, e.g.,
+`Amazon Neptune Database <https://aws.amazon.com/neptune/>`_, as data source for subgraph extraction.
+
+Once the subgraph is extracted, you will need to prepare it as the payload of different APIs for `invoke 
+models for real-time inference
+<https://docs.aws.amazon.com/sagemaker/latest/dg/realtime-endpoints-test-endpoints.html#realtime-endpoints-test-endpoints-api>`_.
+GraphStorm defines a :ref:`specification of the payload contents <rt-request_payload-spec>` for your reference.
+
+Below is an example payload JSON object for node classification inference.
+
+.. code:: yaml
+
+    {
+        "version": "gs-realtime-v0.1",
+        "gml_task": "node_classification",
+        "graph": {
+            "nodes": [
+                {
+                    "node_type": "author",
+                    "node_id": "a4444",
+                    "features": {"feat_num" : [ 0.011269339360296726, ......, ],
+                                 "feat_cat" : "UK"},
+                },
+                {
+                    "node_type": "author",
+                    "node_id": "a39",
+                    "features": {"feat": [-0.0032965524587780237, ......, ],
+                                "feat_cat" : "USA"},
+                },
+                ......
+            ],
+            "edges": [
+                {
+                    "edge_type": [
+                        "author",
+                        "writing",
+                        "paper"
+                    ],
+                    "src_node_id": "p4463",
+                    "dest_node_id": "p4463",
+                    "features": {"feat1": [ 1.411269339360296726, ......, ]},
+                                 "feat2" : "1980s"},
+                },
+                ......
+            ]
+        },
+        "targets": [
+            {
+                "node_type": "author",
+                "node_id": "a39"
+            }
+        ]
+    }
+
+Invoke endpoints
+****************
+There are multiple ways to invoke a Sagemaker real-time inference endpoint as documented in
+`SageMaker Developer Guide <https://docs.aws.amazon.com/sagemaker/latest/dg/realtime-endpoints-test-endpoints.html#realtime-endpoints-test-endpoints-api>`_.
+
+.. note:: GraphStorm real-time inference endpoints currently only support "application/json" as content type.
+
+Here is an example of how you can read a payload from a JSON file and use the boto3 APIs to
+invoke an endpoint.
+
+.. code-block:: python
+
+    import boto3
+    import json
+
+    # Create a SageMaker client object\n",
+    sagemaker = boto3.client('sagemaker')
+    # Create a SageMaker runtime client object using your IAM role ARN\n",
+    runtime = boto3.client('sagemaker-runtime',
+                           aws_access_key_id='your access key string',
+                           aws_secret_access_key='your secret key string',
```

Comment:
Let's remove these, it's safer to let boto3 find credentials on its own rather than having them in code
```suggestion
```